### PR TITLE
Update node /etc/updatedb.conf in csi plugin

### PR DESCRIPTION
**Pull request by abowloflrf**
Signed-off-by: Ruofeng Lei <ruofenglei@outlook.com>

### Ⅰ. Describe what this PR does

#2058

Add a plugin in CSI to modify `/etc/updatedb.conf` on host node whiling starting the CSI Pod. The changes are only made when `/etc/updatedb.conf` do exist on the node.

The fs to prune in `PRUNEFS` use `runtime.mountRoot` in `values.yaml`.
The paths to prune in `PRUNEPATHS` use `csi.pruneFs` in `values.yaml`.

Which add the flowing flags in CSI plugin container:

```
- --prune-fs=fuse.alluxio-fuse,fuse.jindofs-fuse,JuiceFS,fuse.goosefs-fuse,ossfs
- --prune-path=/runtime-mnt
```

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/csi/updatedbconf/register.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+	"github.com/golang/glog"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/fluid-cloudnative/fluid/pkg/csi/config"
+)
+
+// Register update the host /etc/updatedb.conf
+func Register(_ manager.Manager, cfg config.Config) error {
+	content, err := os.ReadFile(updatedbConfPath)
```
Comment (cheyang) [inline pkg/csi/updatedbconf/register.go:14]: 是否可以做成幂等？如果文件内容符合预期就不会有写操作。

File: cmd/csi/app/csi.go [hunk 2]
```diff
@@ -75,6 +77,8 @@ func init() {
 		ErrorAndExit(err)
 	}
 
+	startCmd.Flags().StringSliceVarP(&pruneFs, "prune-fs", "", []string{}, "Prune fs to add in /etc/updatedb.conf, separated by comma")
```
Comment (cheyang) [inline cmd/csi/app/csi.go:80]: 可以设置pruneFs的默认值下。

File: pkg/csi/updatedbconf/updatedbconf.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+)
+
+const (
+	updatedbConfPath    = "/host-etc/updatedb.conf"
+	configKeyPruneFs    = "PRUNEFS"
+	configKeyPrunePaths = "PRUNEPATHS"
+)
+
+// updateLine add new config items to a line
+func updateLine(line string, key string, values []string) string {
```
Comment (cheyang) [inline pkg/csi/updatedbconf/updatedbconf.go:15]: 是否可以返回一个shouldUpdate的bool值？

File: charts/fluid/fluid/values.yaml [hunk 4]
```diff
@@ -24,6 +24,7 @@ csi:
     image: fluidcloudnative/fluid-csi:v0.8.0-7dd9d35
   kubelet:
     rootDir: /var/lib/kubelet
+  pruneFs: fuse.alluxio-fuse,fuse.jindofs-fuse,JuiceFS,fuse.goosefs-fuse,ossfs
```
Comment (zwwhdls) [inline charts/fluid/fluid/values.yaml:27]: "fuse.juicefs" instead of JuiceFS

File: pkg/csi/updatedbconf/register.go (hunk 1 above)
Comment (abowloflrf) [reply pkg/csi/updatedbconf/register.go:14]: done, if the returned `newcontent` has no changes, skip write the config file.

File: cmd/csi/app/csi.go (hunk 2 above)
Comment (abowloflrf) [reply cmd/csi/app/csi.go]: done

File: pkg/csi/updatedbconf/updatedbconf.go (hunk 3 above)
Comment (abowloflrf) [reply pkg/csi/updatedbconf/updatedbconf.go]: done

File: charts/fluid/fluid/values.yaml (hunk 4 above)
Comment (abowloflrf) [reply charts/fluid/fluid/values.yaml]: done

**Review by cheyang — APPROVED** [review]

**Pull request merged**